config/src/main/java/org/springframework/security/config/http/OAuth2ResourceServerBeanDefinitionParser.java

```diff
@@ -0,0 +1,336 @@
+				.rootBeanDefinition(BearerTokenRequestMatcher.class);
+		requestMatcherBuilder.addConstructorArgValue(bearerTokenResolver);
+		BeanDefinition requestMatcher = requestMatcherBuilder.getBeanDefinition();
+
+		this.entryPoints.put(requestMatcher, this.authenticationEntryPoint);
+		this.deniedHandlers.put(requestMatcher, this.accessDeniedHandler);
+		this.ignoreCsrfRequestMatchers.add(requestMatcher);
+
+		BeanDefinitionBuilder filterBuilder = BeanDefinitionBuilder
+				.rootBeanDefinition(BearerTokenAuthenticationFilter.class);
+		BeanMetadataElement authenticationManagerResolver = getAuthenticationManagerResolver(oauth2ResourceServer);
+		filterBuilder.addConstructorArgValue(authenticationManagerResolver);
+		filterBuilder.addPropertyValue(BEARER_TOKEN_RESOLVER, bearerTokenResolver);
+		filterBuilder.addPropertyValue(AUTHENTICATION_ENTRY_POINT, this.authenticationEntryPoint);
+		return filterBuilder.getBeanDefinition();
+	}
+
+	void validateConfiguration(Element oauth2ResourceServer, Element jwt, Element opaqueToken, ParserContext pc) {
+		if (!oauth2ResourceServer.hasAttribute(AUTHENTICATION_MANAGER_RESOLVER_REF)) {
+			if (jwt == null && opaqueToken == null) {
+				pc.getReaderContext().error
+						("Didn't find authentication-manager-resolver-ref, <jwt>, or <opaque-token>. " +
+								"Please select one.", oauth2ResourceServer);
+			}
+			return;
+		}
+
+		if (jwt != null) {
+			pc.getReaderContext().error
+					("Found <jwt> as well as authentication-manager-resolver-ref. " +
+							"Please select just one.", oauth2ResourceServer);
+		}
+
+		if (opaqueToken != null) {
+			pc.getReaderContext().error
+					("Found <opaque-token> as well as authentication-manager-resolver-ref. " +
+							"Please select just one.", oauth2ResourceServer);
+		}
+	}
+
+	BeanMetadataElement getAuthenticationManagerResolver(Element element) {
+		String authenticationManagerResolverRef = element.getAttribute(AUTHENTICATION_MANAGER_RESOLVER_REF);
+		if (!StringUtils.isEmpty(authenticationManagerResolverRef)) {
+			return new RuntimeBeanReference(authenticationManagerResolverRef);
+		}
+		BeanDefinitionBuilder authenticationManagerResolver = BeanDefinitionBuilder
+				.rootBeanDefinition(StaticAuthenticationManagerResolver.class);
+		authenticationManagerResolver.addConstructorArgValue(this.authenticationManager);
+		return authenticationManagerResolver.getBeanDefinition();
+	}
+
+	BeanMetadataElement getBearerTokenResolver(Element element) {
+		String bearerTokenResolverRef = element.getAttribute(BEARER_TOKEN_RESOLVER_REF);
+		if (StringUtils.isEmpty(bearerTokenResolverRef)) {
+			return new RootBeanDefinition(DefaultBearerTokenResolver.class);
+		} else {
+			return new RuntimeBeanReference(bearerTokenResolverRef);
+		}
+	}
+}
+
+final class JwtBeanDefinitionParser implements BeanDefinitionParser {
+	static final String DECODER_REF = "decoder-ref";
+	static final String JWK_SET_URI = "jwk-set-uri";
```

Comment:
Let's rename the attribute to `jwkset-uri` - NOTE: This is how it's named in `<client-registration>` element

Reply:
On the DSL, this is called `jwkSetUri`, and `jwkSetUri` appears to be the prevailing case pattern in variable and method names. The most natural translation I would think would be to place a dash wherever the case changes.

Based on this, I'd recommend changing `<client-registration>` to be `jwk-set-uri` so that it aligns with the DSL.

Reply:
That makes sense. I'll update to `jwk-set-uri`  for `<client-registration>`